app/scripts/metamask-controller.js

```diff
@@ -6204,9 +6205,11 @@ export default class MetamaskController extends EventEmitter {
         // network configuration-related
         setActiveNetwork: async (networkClientId) => {
           await this.networkController.setActiveNetwork(networkClientId);
-          // if the origin has the eth_accounts permission
-          // we set per dapp network selection state
+
+          // If the origin has the `eth_accounts` permission, or if it's a Snap,
+          // we set per origin network selection state.
           if (
+            subjectType === SubjectType.Snap ||
```

Comment:
What is the reasoning for this?

Reply:
`setActiveNetwork` checks if the origin has the `eth_accounts` permission to see if the network should be set for that specific origin. Snaps may not (cannot?) have this permission, but we want to use the per-origin-network regardless for Snaps.